ipam/mas.go

```diff
@@ -5,129 +5,193 @@ package ipam
 
 import (
 	"encoding/json"
+	"io/ioutil"
 	"net"
-	"net/http"
-	"time"
+	"runtime"
+	"strings"
 
-	"github.com/Azure/azure-container-networking/common"
 	"github.com/Azure/azure-container-networking/log"
 )
 
 const (
-	// Host URL to query.
-	masQueryUrl = "http://169.254.169.254:6642/ListNetwork"
-
-	// Minimum time interval between consecutive queries.
-	masQueryInterval = 10 * time.Second
+	defaultLinuxFilePath = "/etc/kubernetes/interfaces.json"
+	defaultWindowsFilePath = `c:\k\interfaces.json`
 )
 
 // Microsoft Azure Stack IPAM configuration source.
 type masSource struct {
-	name          string
-	sink          addressConfigSink
-	queryUrl      string
-	queryInterval time.Duration
-	lastRefresh   time.Time
+	name       string
+	sink       addressConfigSink
+	fileLoaded bool
+	filePath   string
 }
 
 // MAS host agent JSON object format.
-type jsonObject struct {
-	Isolation string
-	IPs       []struct {
-		IP              string
-		IsolationId     string
-		Mask            string
-		DefaultGateways []string
-		DnsServers      []string
-	}
+type NetworkInterfaces struct {
+	Interfaces []Interface
+}
+
+type Interface struct {
+	MacAddress string
+	IsPrimary  bool
+	IPSubnets  []IPSubent
+}
+
+type IPSubent struct {
+	Prefix      string
+	IPAddresses []IPAddress
+}
+
+type IPAddress struct {
+	Address   string
+	IsPrimary bool
 }
 
 // Creates the MAS source.
 func newMasSource(options map[string]interface{}) (*masSource, error) {
-	queryUrl, _ := options[common.OptIpamQueryUrl].(string)
-	if queryUrl == "" {
-		queryUrl = masQueryUrl
-	}
-
-	i, _ := options[common.OptIpamQueryInterval].(int)
-	queryInterval := time.Duration(i) * time.Second
-	if queryInterval == 0 {
-		queryInterval = masQueryInterval
+	var filePath string
+	if runtime.GOOS == "windows" {
+		filePath = defaultWindowsFilePath
+	} else {
+		filePath = defaultLinuxFilePath
 	}
 
 	return &masSource{
-		name:          "MAS",
-		queryUrl:      queryUrl,
-		queryInterval: queryInterval,
+		name: "MAS",
+		filePath: filePath,
 	}, nil
 }
 
 // Starts the MAS source.
-func (s *masSource) start(sink addressConfigSink) error {
-	s.sink = sink
+func (source *masSource) start(sink addressConfigSink) error {
```

Comment:
there's no need to return anything if it is always nil

Reply:
`masSource` implements the `addressConfigSource` interface (https://github.com/Azure/azure-container-networking/blob/master/ipam/manager.go#L51-L55) and so if I change the method signature it will no longer satisfy the interface.